src/app/_components/SortListbox.tsx

```diff
@@ -1,55 +1,46 @@
 'use client'
-import { Fragment, useState } from 'react'
+import { Fragment } from 'react'
 
 import { Listbox } from '@headlessui/react'
 import { ArrowsDownUp, CaretDown, Check } from '@phosphor-icons/react/dist/ssr'
 import clsx from 'clsx'
 
 import { Icon } from '@/components/Icon'
 
-export enum SortType {
-  Newest = 'newest',
-  Oldest = 'oldest',
-}
+import { type SortOptionType } from '@/types/sortTypes'
 
-type SortSetting = {
-  id: SortType
-  name: string
-}
+import { sortSettings } from '@/constants/sortConstants'
 
 type SortListboxProps = {
-  onSortTypeChange: (selectedOption: SortType) => void
+  sortOption: SortOptionType
+  onSortOptionChange: (selectedSortOption: SortOptionType) => void
 }
 
-const sortSettings: SortSetting[] = [
-  { id: SortType.Newest, name: 'Newest' },
-  { id: SortType.Oldest, name: 'Oldest' },
-]
-
-export function SortListbox({ onSortTypeChange }: SortListboxProps) {
-  const [selectedSortType, setSelectedSortType] = useState(sortSettings[0])
-
-  function handleChange(newSortType: SortSetting) {
-    setSelectedSortType(newSortType)
-    onSortTypeChange(newSortType.id)
-  }
+export function SortListbox({
+  sortOption,
+  onSortOptionChange,
+}: SortListboxProps) {
+  const selectedSortSetting =
+    sortSettings.find((option) => option.id === sortOption) || sortSettings[0]
 
   return (
-    <Listbox value={selectedSortType} onChange={handleChange}>
+    <Listbox value={sortOption} onChange={onSortOptionChange}>
       {() => (
         <>
```

Comment:
```suggestion
```

This seems unnecessary; can we remove it?

Reply:
Yes! Leftover. 🙏🏼